src/web/mock.rs

```diff
@@ -53,39 +53,37 @@ pub(crate) async fn server(context: MockServerContext) -> Result<(), Error> {
     server.await.context("Error on web server execution")
 }
 
-async fn handler<T>(req: Request<T>, state: SharedState) -> Result<Response<Body>, Error> {
-    match search_for_mock(&req, state).await? {
+async fn handler(req: Request<Body>, state: SharedState) -> Result<Response<Body>, Error> {
+    let (parts, body) = req.into_parts();
+    let content = hyper::body::to_bytes(body).await?;
+
+    let incoming_request = IncomingRequest {
+        method: HttpMethod::try_from(parts.method)?,
+        path: parts.uri.path().to_string(),
+        body: String::from_utf8(content.to_vec())?, //TODO read header to use the good string encoding
```

Comment:
Suggestion :
Adding the String directly make a test to parse the body in the target for each expectation that could be slow.
Maybe we could add an enum with all managed types of body in incoming request and try based on `Content-Type` here and let an unknown type to try on each expectation.
It's more difficult but could be interesting

Could also another issue and PR

Reply:
Good point, but I think we can reduce cost of expectation testing by improving the algorithm in a really simple manner : test method and path first will reduce drastically number of test to perform on body content and would be enough IMO.

I opened an issue about this problem #66